src/core/createCookie.js

```diff
@@ -9,125 +9,53 @@ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTA
 OF ANY KIND, either express or implied. See the License for the specific language
 governing permissions and limitations under the License.
 */
-import cookie from "../utils/cookie";
-import cookieDetails from "../constants/cookieDetails";
-import getTopLevelCookieDomain from "../utils/getTopLevelCookieDomain";
-
-const { ALLOY_COOKIE_NAME, ALLOY_COOKIE_TTL } = cookieDetails;
-const cookieProxyByPropertyId = {};
-
-const safeJSONParse = (object, cookieName) => {
-  try {
-    return JSON.parse(object);
-  } catch (error) {
-    throw new Error(`Invalid cookie format in ${cookieName} cookie`);
-  }
-};
-
-/**
- * The purpose of this proxy is to cache the cookie so we don't have to
- * read and deserialize it every time a piece of it is accessed.
- */
-const createCookieProxy = (propertyId, cookieDomain = "") => {
-  // Because the cookie proxy caches the deserialized cookie, we want to make
-  // sure there's only one cookie proxy created per cookie because multiple
-  // caches per cookies would result in caches getting stale.
-  let cookieProxy = cookieProxyByPropertyId[propertyId];
-
-  if (cookieProxy) {
-    return cookieProxy;
-  }
-
-  const cookieName = `${ALLOY_COOKIE_NAME}_${propertyId}`;
-  let deserializedCookie;
-  let cookieHasBeenRead = false;
-
-  cookieProxy = {
-    get() {
-      // We don't read the cookie right when the cookie proxy is created
-      // because we don't know if the user has opted in. If the user
-      // hasn't opted in, we're legally obligated to not read the cookie.
-      // If a component tries to read something off the cookie though,
-      // we assume that the component has received word that the user
-      // has opted-in. The responsibility is on the component.
-      if (cookieHasBeenRead) {
-        return deserializedCookie;
-      }
-
-      const serializedCookie = cookie.get(cookieName);
-      deserializedCookie =
-        serializedCookie && safeJSONParse(serializedCookie, cookieName);
-      cookieHasBeenRead = true;
-      return deserializedCookie;
-    },
-    set(updatedCookie) {
-      deserializedCookie = updatedCookie;
-      cookie.set(cookieName, updatedCookie, {
-        expires: ALLOY_COOKIE_TTL,
-        domain: cookieDomain || getTopLevelCookieDomain(window, cookie)
-      });
-    }
-  };
-
-  cookieProxyByPropertyId[propertyId] = cookieProxy;
-
-  return cookieProxy;
-};
-
-// TODO: Support passing a configurable expiry in the config when creating this cookie.
-const createCookie = (componentNamespace, propertyId, cookieDomain = "") => {
-  const cookieProxy = createCookieProxy(propertyId, cookieDomain);
 
+export default (cookieProxy, componentNamespace) => {
```

Comment:
These names are somewhat confusing.  Having a cookie object and a cookieProxy object, I would assume the cookieProxy wraps the cookie Object.  Perhaps a way to solve this potential confusion is to rename the cookie object to `namespacedCookie`.  Indeed, the main purpose of the createCookie function is to add namespace functionality to the cookieProxy.  This would align nicely with the `createNamespacedStorage` function as well.

Reply:
Yeah, I like that. I'll make the change.

Reply:
Actually, I'm going to do this in a separate PR. Thanks for the feedback.